posts/2021-05-17-metrics-jee9-21006-beta.adoc

```diff
@@ -0,0 +1,178 @@
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/austin0
+author_github: https://github.com/austin0
+seo-title: Request Timing Metrics and new Jakarta EE9 support in Open Liberty 21.0.0.6-beta - OpenLiberty.io
+seo-description: Open Liberty 21.0.0.6-beta now supports integrating Request Timing information with MicroProfile Metrics, along with a number of new Jakarta EE9 supporting features including OAuth/SSO authorization and automatic certificate management.
+blog_description: Open Liberty 21.0.0.6-beta now supports integrating Request Timing information with MicroProfile Metrics, along with a number of new Jakarta EE9 supporting features including OAuth/SSO authorization and automatic certificate management.
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+---
+= Request Timing Metrics and new Jakarta EE9 support in Open Liberty 21.0.0.6-beta
+Austin Bailey <https://github.com/austin0>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+
+Open Liberty 21.0.0.6-beta bring improvements to the MicroProfile Metrics feature, allowing for information gathered via the Request Timing feature to be displayed on the metrics endpoint. Also included are a number of new Jakarta EE9 supporting features, including OAuth/SSO authorization and automatic certificate management.
+
+We have two beta packages for link:{url-about}[Open Liberty]:
+
+* <<allbeta, All Beta Features>>: a larger package that contains all Open Liberty beta features (including Jakarta EE 9 beta features) and GA features and functions.
+* <<jakarta, Jakarta EE 9 Beta Features>>: a lightweight package that contains only the Jakarta EE 9 features.
+
+This means that you can now try out our in-development Open Liberty features by just adding the relevant coordinates to your build tools.
+
+If you try either package, <<feedback, let us know what you think>>.
+[#allbeta]
+== All Beta Features package
+
+The All Beta Features package includes the following beta features:
+
+* <<metrics, Request Timing now supported by MicroProfile Metrics>>
+
+[#metrics]
+=== Request Timing now supported by MicroProfile Metrics
+
+The request timing feature is used to keep track of the slow and hung requests for servlet requests with a `RequestTimingStats` MXBean. The Microprofile Metrics features on the other hand will provide vendor metrics that are distinct to the Open Liberty runtime. These vendor metrics are typically sourced from the MXBeans provided by various components in the Open Liberty runtime. However, until now the two have not crossed paths. 
+
+Starting with the Open Liberty 21.0.0.6-beta the `RequestTimingStats` MXBean will now have it's data retrieved by the MicroProfile Metrics feature for reporting on the `/metrics` (or `/metrics/vendor`) endpoint. This functionality is compatible with both the 2.X and 3.X MicroProfile Metrics features.
+
+Below is a sample output of the new request timing metrics:
+
+[source, xml]
```

Comment:
For the release blog post (not necessary for this beta), it would be good if we could get the data visualized in an image, so something like Grafana

Reply:
@Channyboy Would it be possible to provide an image for the GA release blog?